src/pages/AccountSettings/tabs/OktaAccess/OktaConfigForm/OktaConfigForm.tsx

```diff
@@ -0,0 +1,180 @@
+  redirectUri: z.string().url('Redirect URI must be a valid URL'),
+})
+
+type FormValues = z.infer<typeof FormSchema>
+
+export function OktaConfigForm() {
+  const { register, handleSubmit, formState } = useForm<FormValues>({
+    resolver: zodResolver(FormSchema),
+    mode: 'onChange',
+  })
+
+  const [oktaEnabled, setOktaEnabled] = useState(false)
+  const [oktaLoginEnforce, setOktaLoginEnforce] = useState(false)
+  const [showPassword, setShowPassword] = useState(false)
+
+  const onSubmit: SubmitHandler<FormValues> = (data) => {
+    console.log('Form Data: ', data)
+  }
+
+  return (
+    <div className="w-5/6 border-2 border-solid border-ds-gray-primary p-4 text-ds-gray-octonary">
+      <form
+        onSubmit={handleSubmit(onSubmit)}
+        className="mb-4 flex flex-col gap-4"
+      >
+        <div className="flex flex-col gap-1">
+          <h2 className="text-base font-semibold">Step 1: Enable Okta Sync</h2>
+          <p>
+            To connect Codecov with Okta, you need to enable the
+            synchronization. Please enter the necessary Okta credentials below
+            and toggle the sync option to start the synchronization process.
+          </p>
+        </div>
+        <div className="flex flex-col gap-4 bg-ds-gray-primary p-4">
+          <div className="flex flex-col gap-1">
+            <label htmlFor="clientId" className="block font-semibold">
+              Client ID
+            </label>
+            <TextInput
+              {...register('clientId', {
+                required: true,
+              })}
+              type="text"
+              id="clientId"
+              placeholder="Enter Client ID"
+            />
+            {formState.errors.clientId && (
+              <p className="mt-1 text-codecov-red">
+                {formState.errors.clientId.message}
+              </p>
+            )}
+          </div>
+          <div className="flex flex-col gap-1">
+            <label htmlFor="clientSecret" className="block font-semibold">
+              Client Secret
+            </label>
+            <div className="relative">
+              <TextInput
+                {...register('clientSecret', { required: true })}
+                type={showPassword ? 'text' : 'password'}
+                id="clientSecret"
+                placeholder="Enter Client Secret"
+              />
+              <button
+                type="button"
+                onClick={() => setShowPassword(!showPassword)}
+                className="absolute inset-y-0 right-0 flex items-center pr-3 text-sm leading-5"
+              >
+                <Icon
+                  name={showPassword ? 'eye' : 'eyeOff'}
+                  size="sm"
+                  variant="solid"
+                />
+              </button>
+            </div>
+            {formState.errors.clientSecret && (
+              <p className="mt-1 text-codecov-red">
+                {formState.errors.clientSecret.message}
+              </p>
+            )}
```

Comment:
`m`: Can we swap these to ternary checks: 

```jsx
            {formState.errors.clientSecret ? (
              <p className="mt-1 text-codecov-red">
                {formState.errors.clientSecret.message}
              </p>
            ) : null}
```

They add in some extra protections that double `&` do not, if you want you can read more about these [here](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx).